e2e/shannon_relay_test.go

```diff
@@ -4,102 +4,96 @@ package e2e
 
 import (
 	"bytes"
-	"context"
 	"encoding/json"
 	"errors"
 	"fmt"
 	"io"
-	"math/rand"
 	"net/http"
-	"net/url"
+	"os"
 	"testing"
-	"time"
 
-	"github.com/pokt-network/poktroll/pkg/polylog"
-	"github.com/pokt-network/poktroll/pkg/polylog/polyzero"
 	"github.com/stretchr/testify/require"
-
-	"github.com/buildwithgrove/path/config"
-	"github.com/buildwithgrove/path/relayer"
-	"github.com/buildwithgrove/path/relayer/shannon"
-	"github.com/buildwithgrove/path/request"
 )
 
-const configPath = ".config.test.yaml"
+// localdev.me is a hosted domain that resolves to 127.0.0.1 (localhost).
+// This allows a subdomain to be specified without modifying /etc/hosts.
+// It is hosted by AWS. See https://codeengineered.com/blog/2022/localdev-me/
+const localdevMe = "localdev.me"
 
-var testTimeout = 20 * time.Second
+// When the ephemeral PATH Docker container is running it exposes a dynamically
+// assigned port. This global variable is used to capture the port number.
+var pathPort string
+
+func TestMain(m *testing.M) {
+	// Initialize the ephemeral PATH Docker container
+	pool, resource, containerPort := setupPathDocker()
+
+	// Assign the port the container is listening on to the global variable
+	pathPort = containerPort
+
+	// Run PATH E2E Shannon relay tests
+	exitCode := m.Run()
+
+	// Cleanup the ephemeral PATH Docker container
+	cleanupPathDocker(m, pool, resource)
+
+	// Exit with the test result
+	os.Exit(exitCode)
+}
 
-// TODO_INCOMPLETE: add an action to the CI for running E2E tests, which at the minimum
-// includes using the e2e tag and defining secrets to be used as environment variables, e.g. gateway/signer private key and address.
-//
 // TODO_IMPROVE: use gocuke (github.com/regen-network/gocuke) for defining and running E2E tests.
```

Comment:
@adshmh @Olshansk is this still a `TODO`? I don't know enough about `gocuke` to say for sure what the advantage is but seems like it's probably not needed at this point, WDYT?

Reply:
Let's remove it for now. 

It'll come back once we move to natural language E2E tests: https://github.com/pokt-network/poktroll/blob/main/e2e/tests/0_settlement.feature